packages/replay/src/uploadWorker.ts

```diff
@@ -0,0 +1,56 @@
+import type { URL } from "url";
+import type { AgentOptions } from "http";
+
+import { parentPort } from "worker_threads";
+import fetch from "node-fetch";
+import { Agent as HttpAgent } from "http";
+import { Agent as HttpsAgent } from "https";
+import { getUserAgent } from "./utils";
+
+// TODO: Swap out console.log with DEBUG logs, but it's worker thread
+// and current dbg log implementation will race parent thread and worker thread
+
+const agentConfig: AgentOptions = {
```

Comment:
We accept an agent config for upload already (e.g. to support using a proxy) and should probably merge that configuration in as here well.

Reply:
We have the `Agent` but we can't pass the instance of the class over to the Worker thread, parameters need to be serializable. We can refactor to access `AgentOptions` instead and initialize the class ourselves. I was thinking of doing that separately, not in this PR.

Reply:
Okay. Let's open the ticket for that and add a `//TODO` here referencing it.